lms/djangoapps/monitoring/scripts/process_cookie_monitoring_logs.py

```diff
@@ -0,0 +1,212 @@
+            continue
+        match = cookie_log_regex.search(raw_cookie_log)
+        if not match:
+            logging.error("Malformed cookie entry. Skipping row.")
+            continue
+
+        cookie_header_size = int(match.group("total"))
+        if cookie_header_size == 0:
+            continue
+
+        cookie_sizes_str = match.group("cookie_sizes").strip()
+
+        cookie_sizes = cookie_sizes_str.split(", ")
+        for cookie_size in cookie_sizes:
+            match = cookie_size_regex.search(cookie_size)
+            if not match:
+                logging.error(f"Could not parse cookie size from: {cookie_size}")
+                continue
+            cookie_header_sizes[match.group("name")] = int(match.group("size"))
+
+        cookie_header_size_computed = max(
+            0, sum(len(name) + size + 3 for (name, size) in cookie_header_sizes.items()) - 2
+        )
+
+        cookie_headers.append({
+            "datetime": parser.parse(row.get("_time")),
+            "env": row.get("index"),
+            "cookie_header_size": cookie_header_size,
+            "cookie_header_size_computed": cookie_header_size_computed,
+            "cookie_sizes": cookie_header_sizes,
+        })
+
+    return cookie_headers
+
+
+def process_cookie_headers(cookie_headers):
+    """
+    Process the parsed cookie header log entries.
+
+    Arguments:
+        cookie_headers: a list of dicts containing parsed details.
+
+    Returns a dict of processed cookies.
+    """
+    processed_cookies = {}
+    for cookie_header in cookie_headers:
+        for (name, size) in cookie_header["cookie_sizes"].items():
+
+            # Replace parameterized cookies. For example:
+            #   _hjSessionUser_111111 => _hjSessionUser_{id}
+            for (regex, replacement_name) in PARAMETERIZED_COOKIES:
+                if regex.fullmatch(name):
+                    logging.debug(f"Replacing {name} with {replacement_name}.")
+                    name = replacement_name
+                    break
+
+            processed_cookie = processed_cookies.get(name, {})
+
+            # compute the full size each cookie takes up in the cookie header, including name and delimiters
+            full_size = len(name) + size + 3
+            set_max_attribute(processed_cookie, "max_full_size", full_size)
+            set_min_attribute(processed_cookie, "min_full_size", full_size)
+
+            set_max_attribute(processed_cookie, "max_size", size)
+            set_min_attribute(processed_cookie, "min_size", size)
+
+            processed_cookie["count"] = processed_cookie.get("count", 0) + 1
+
+            # Note: The following details relate to the header, and not to the specific cookie.
+            #   This may give a quick view of cookies associated with the largest header, or the
+            #   header with the most cookies.
+
+            set_max_attribute(processed_cookie, "last_seen", cookie_header["datetime"])
```

Comment:
[inform] @timmc-edx: I made two final changes:
1. I added "last_seen".
2. I added `set_max_attribute` and `set_min_attribute` to minimize the chance of errors due to typos by minimizing repetition.